android-utils/src/main/java/com/virgilsecurity/sdk/androidutils/storage/AndroidKeyStorage.java

```diff
@@ -0,0 +1,543 @@
+     * when {@code authenticationRequired} is {@code true} and key validity duration has been expired.
+     * You have to re-authenticate user, please see the link:
+     * https://developer.android.com/training/articles/keystore#UserAuthentication
+     * @throws KeyStorageException when some error occurred while initializing AndroidKeyStorage.
+     */
+    public static AndroidKeyStorage getInstance(String alias, int keyValidityDuration, String rootPath) {
+        return new AndroidKeyStorage(alias, true, keyValidityDuration, rootPath);
+    }
+
+    /**
+     * Instantiates AndroidKeyStorage class. {@code authenticationRequired} is automatically true, because in other case
+     * {@code keyValidityDuration} makes no sense.
+     *
+     * @param alias               is an alias with which current keystore will be saved.
+     * @param keyValidityDuration default duration is 5 minutes. You can specify other duration of key validity in
+     *                            seconds. After the time specified has expired user has to be re-authenticated.
+     *                            You can set requirement that every key usage will need authentication, for details
+     *                            please, see {@link KeyGenParameterSpec.Builder#setUserAuthenticationRequired} docs.
+     *
+     * @return AndroidKeyStorage instance.
+     *
+     * @throws com.virgilsecurity.sdk.androidutils.exception.UserNotAuthenticatedException
+     * when {@code authenticationRequired} is {@code true} and key validity duration has been expired.
+     * You have to re-authenticate user, please see the link:
+     * https://developer.android.com/training/articles/keystore#UserAuthentication
+     * @throws KeyStorageException when some error occurred while initializing AndroidKeyStorage.
+     */
+    public static AndroidKeyStorage getInstance(String alias, int keyValidityDuration) {
+        return new AndroidKeyStorage(alias, true, keyValidityDuration,
+                System.getProperty("user.home"));
+    }
+
+    /**
+     * Instantiates AndroidKeyStorage class. Default key validity time (after which user has to be re-authenticated)
+     * is 5 minutes.
+     *
+     * @param alias    is an alias with which current keystore will be saved.
+     * @param rootPath path which will be used to store keys.
+     *
+     * @return AndroidKeyStorage instance.
+     *
+     * @throws com.virgilsecurity.sdk.androidutils.exception.UserNotAuthenticatedException
+     * when {@code authenticationRequired} is {@code true} and key validity duration has been expired.
+     * You have to re-authenticate user, please see the link:
+     * https://developer.android.com/training/articles/keystore#UserAuthentication
+     * @throws KeyStorageException when some error occurred while initializing AndroidKeyStorage.
+     */
+    public static AndroidKeyStorage getInstance(String alias, String rootPath) {
```

Comment:
Why do we need a static method? Let's use constructors instead

Reply:
Or Builder

Reply:
Static method, because it throws exceptions. I think that constructors that throw exceptions are not the best approach. Maybe Builder is better. Or you think that constructors with exceptions are ok?